Type issues in hw/misc/trace.h

It seems that `hw/misc/trace.h` is not cross-compatible due to some of the types defined in it.
The build error occured on kernel `5.4.0-52-generic` on Ubuntu `20.04` with GCC `9.3.0`.

```
In file included from hw/misc/trace.c:5:
hw/misc/trace.h: In function ‘_nocheck__trace_hermes_dma_desc’:
hw/misc/trace.h:4163:170: error: stray ‘\’ in program
 4163 | l: 0x%"PRIx32" len: 0x%"PRIx32" src_addr: 0x%"PRIx64" dst_addr: 0x%"PRIx64" nxt_addr: 0x%"PRIx64" "\n",
      |                                                                                                    ^

hw/misc/trace.h:4163:49: error: expected ‘)’ before ‘n’
 4163 |         qemu_log("%d@%zu.%06zu:hermes_dma_desc " "Hermes DMA Desc: ctrl: 0x%"PRIx32" len: 0x%"PRIx32" src_addr: 0x%"PRIx64" dst_addr: 0x%"PRIx64" nxt_addr: 0x%"PRIx64" "\n",
      |                                                 ^                                                                                                                         ~
      |                                                 )
hw/misc/trace.h:4163:172: error: missing terminating " character [-Werror]
 4163 |  0x%"PRIx32" len: 0x%"PRIx32" src_addr: 0x%"PRIx64" dst_addr: 0x%"PRIx64" nxt_addr: 0x%"PRIx64" "\n",
      |                                                                                                    ^

hw/misc/trace.h:4163:172: error: missing terminating " character
 4163 |  0x%"PRIx32" len: 0x%"PRIx32" src_addr: 0x%"PRIx64" dst_addr: 0x%"PRIx64" nxt_addr: 0x%"PRIx64" "\n",
      |                                                                                                    ^~

hw/misc/trace.h:4163:20: error: format ‘%d’ expects a matching ‘int’ argument [-Werror=format=]
 4163 |         qemu_log("%d@%zu.%06zu:hermes_dma_desc " "Hermes DMA Desc: ctrl: 0x%"PRIx32" len: 0x%"PRIx32" src_addr: 0x%"PRIx64" dst_addr: 0x%"PRIx64" nxt_addr: 0x%"PRIx64" "\n",
      |                   ~^
      |                    |
      |                    int
hw/misc/trace.h:4163:24: error: format ‘%zu’ expects a matching ‘size_t’ argument [-Werror=format=]
 4163 |         qemu_log("%d@%zu.%06zu:hermes_dma_desc " "Hermes DMA Desc: ctrl: 0x%"PRIx32" len: 0x%"PRIx32" src_addr: 0x%"PRIx64" dst_addr: 0x%"PRIx64" nxt_addr: 0x%"PRIx64" "\n",
      |                      ~~^
      |                        |
      |                        long unsigned int
hw/misc/trace.h:4163:30: error: format ‘%zu’ expects a matching ‘size_t’ argument [-Werror=format=]
 4163 |         qemu_log("%d@%zu.%06zu:hermes_dma_desc " "Hermes DMA Desc: ctrl: 0x%"PRIx32" len: 0x%"PRIx32" src_addr: 0x%"PRIx64" dst_addr: 0x%"PRIx64" nxt_addr: 0x%"PRIx64" "\n",
      |                          ~~~~^
      |                              |
      |                              long unsigned int
hw/misc/trace.h:4163:18: error: format ‘%x’ expects a matching ‘unsigned int’ argument [-Werror=format=]
 4163 |         qemu_log("%d@%zu.%06zu:hermes_dma_desc " "Hermes DMA Desc: ctrl: 0x%"PRIx32" len: 0x%"PRIx32" src_addr: 0x%"PRIx64" dst_addr: 0x%"PRIx64" nxt_addr: 0x%"PRIx64" "\n",
      |                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /home/nhed/Development/eid-hermes-qemu/include/qemu/osdep.h:89,
                 from hw/misc/trace.c:3:
/usr/include/inttypes.h:120:19: note: format string is defined here
  120 | # define PRIx32  "x"
In file included from hw/misc/trace.c:5:
hw/misc/trace.h:4163:18: error: format ‘%x’ expects a matching ‘unsigned int’ argument [-Werror=format=]
 4163 |         qemu_log("%d@%zu.%06zu:hermes_dma_desc " "Hermes DMA Desc: ctrl: 0x%"PRIx32" len: 0x%"PRIx32" src_addr: 0x%"PRIx64" dst_addr: 0x%"PRIx64" nxt_addr: 0x%"PRIx64" "\n",
      |                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /home/nhed/Development/eid-hermes-qemu/include/qemu/osdep.h:89,
                 from hw/misc/trace.c:3:
/usr/include/inttypes.h:120:19: note: format string is defined here
  120 | # define PRIx32  "x"
In file included from hw/misc/trace.c:5:
hw/misc/trace.h:4163:18: error: format ‘%lx’ expects a matching ‘long unsigned int’ argument [-Werror=format=]
 4163 |         qemu_log("%d@%zu.%06zu:hermes_dma_desc " "Hermes DMA Desc: ctrl: 0x%"PRIx32" len: 0x%"PRIx32" src_addr: 0x%"PRIx64" dst_addr: 0x%"PRIx64" nxt_addr: 0x%"PRIx64" "\n",
      |                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /home/nhed/Development/eid-hermes-qemu/include/qemu/osdep.h:89,
                 from hw/misc/trace.c:3:
/usr/include/inttypes.h:121:34: note: format string is defined here
  121 | # define PRIx64  __PRI64_PREFIX "x"
In file included from hw/misc/trace.c:5:
hw/misc/trace.h:4163:18: error: format ‘%lx’ expects a matching ‘long unsigned int’ argument [-Werror=format=]
 4163 |         qemu_log("%d@%zu.%06zu:hermes_dma_desc " "Hermes DMA Desc: ctrl: 0x%"PRIx32" len: 0x%"PRIx32" src_addr: 0x%"PRIx64" dst_addr: 0x%"PRIx64" nxt_addr: 0x%"PRIx64" "\n",
      |                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /home/nhed/Development/eid-hermes-qemu/include/qemu/osdep.h:89,
                 from hw/misc/trace.c:3:
/usr/include/inttypes.h:121:34: note: format string is defined here
  121 | # define PRIx64  __PRI64_PREFIX "x"
In file included from hw/misc/trace.c:5:
hw/misc/trace.h:4163:18: error: format ‘%lx’ expects a matching ‘long unsigned int’ argument [-Werror=format=]
 4163 |         qemu_log("%d@%zu.%06zu:hermes_dma_desc " "Hermes DMA Desc: ctrl: 0x%"PRIx32" len: 0x%"PRIx32" src_addr: 0x%"PRIx64" dst_addr: 0x%"PRIx64" nxt_addr: 0x%"PRIx64" "\n",
      |                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /home/nhed/Development/eid-hermes-qemu/include/qemu/osdep.h:89,
                 from hw/misc/trace.c:3:
/usr/include/inttypes.h:121:34: note: format string is defined here
  121 | # define PRIx64  __PRI64_PREFIX "x"
cc1: all warnings being treated as errors
make: *** [/home/nhed/Development/eid-hermes-qemu/rules.mak:69: hw/misc/trace.o] Error 1
```

--- a/hw/misc/trace-events
+++ b/hw/misc/trace-events
@@ -222,6 +222,6 @@ pca955x_gpio_change(const char *description, unsigned id, unsigned prev_state, u
 hermes_bar2_read(unsigned int size, uint64_t addr, uint64_t value) "Hermes BAR2 read sz:%u addr:0x%"PRIx64" data:0x%"PRIx64
 hermes_bar2_write(unsigned int size, uint64_t addr, uint64_t value) "Hermes BAR2 write sz:%u addr:0x%"PRIx64" data:0x%"PRIx64
 hermes_dma(const char *channel, uint64_t src, uint64_t dst, uint64_t len) "Hermes DMA %s src: 0x%" PRIx64" dst: 0x%" PRIx64 " len: %lu"
-hermes_dma_desc(uint32_t ctrl, uint32_t len, uint64_t src_addr, uint64_t dst_addr, uint64_t nxt_addr) "Hermes DMA Desc: ctrl: 0x%"PRIx32" len: 0x%"PRIx32" src_addr: 0x%"PRIx64" dst_addr: 0x%"PRIx64" nxt_addr: 0x%"PRIx64"
+hermes_dma_desc(uint32_t ctrl, uint32_t len, uint64_t src_addr, uint64_t dst_addr, uint64_t nxt_addr) "Hermes DMA Desc: ctrl: 0x%"PRIx32" len: 0x%"PRIx32" src_addr: 0x%"PRIx64" dst_addr: 0x%"PRIx64" nxt_addr: 0x%"PRIx64
 hermes_dma_num_adj(unsigned num_adj) "Hermes DMA: num adj: %u"
 hermes_msix_notify(unsigned vector) "Hermes MSI-X notify %u"

